### PR TITLE
Follow up to reauthentication automated test PR

### PR DESCRIPTION
**Pull request by marcaaron**
@tgolen  Just added some extra comments and things to add more context in the tests and also added assertions for Onyx data.

### Details
Follow up to comments made in https://github.com/Expensify/Expensify.cash/pull/1152

### Fixed Issues (Related to PR)
https://github.com/Expensify/Expensify.cash/pull/1152

### Tests
❌

### CODE REVIEW

File: tests/actions/SessionTest.js [hunk 1]
```diff
@@ -37,26 +53,38 @@ test('Authenticate is called with saved credentials when a session expires', ()
...
             expect(authenticateMock.mock.calls.length).toBe(1);
             return waitForPromisesToResolve();
         })
         .then(() => {
+            // Verify that our credentials were saved and that our session data is correct
+            expect(credentials.login).toBe(TEST_USER_LOGIN);
+            expect(credentials.autoGeneratedLogin).not.toBeUndefined();
+            expect(credentials.autoGeneratedPassword).not.toBeUndefined();
+            expect(session.authToken).toBe(TEST_AUTH_TOKEN);
+            expect(session.accountID).toBe(1);
```
Comment (Julesssss) [inline tests/actions/SessionTest.js:79]: Creating a const for `TEST_AUTH_TOKEN` was a nice improvement. Any reason why the same shouldn't be done with `TEST_ACCOUNT_ID`?

Comment (Julesssss) [reply tests/actions/SessionTest.js:79]: *not a blocker

**Review by Julesssss — DISMISSED** [review]

File: tests/actions/SessionTest.js (hunk 1 above)
Comment (marcaaron) [reply tests/actions/SessionTest.js:79]: No reason. Thanks for pointing this out.

**Review by Julesssss — DISMISSED** [review]

**Comment by marcaaron** [comment]
@tgolen simplified this a bunch. Things feel clearer now I think. Lemme know if anything could use explaining.

**Review by Julesssss — APPROVED** [review]
Much clearer to follow, and removing the mocking logic from API.js is a nice improvement.

**Review by tgolen — COMMENTED** [review]
Awesome, this looks really nice! Thanks.

**Pull request merged**